lib/msf/core/exploit/remote/http/manage_engine_adaudit_plus/target_info.rb

```diff
@@ -0,0 +1,151 @@
+# -*- coding: binary -*-
+
+module Msf::Exploit::Remote::HTTP::ManageEngineAdauditPlus::TargetInfo
+  include Msf::Exploit::Remote::HTTP::ManageEngineAdauditPlus::StatusCodes
+  include Msf::Exploit::Remote::HTTP::ManageEngineAdauditPlus::URIs
+
+  # Check is a target is likely a ManageEngine ADAudit Plus app
+  #
+  # @return [Array] Array containing a status code (Integer) and status message (String). If the target is ADAudit Plus, the Array also contains the server response
+  def adaudit_plus_target_check
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path),
+      'method' => 'GET'
+    })
+
+    return [adaudit_plus_status::CONNECTION_FAILED, 'Connection failed'] unless res
+
+    if res.code == 200 && res.body =~ /<title>ADAudit Plus/
+      [adaudit_plus_status::SUCCESS, 'The target appears to be MangeEngine ADAudit Plus', res]
+    else
+      [adaudit_plus_status::UNEXPECTED_REPLY, 'The target does not appear to be MangeEngine ADAudit Plus']
+    end
+  end
+
+  # Extract the configured aliases for configured active directory domains from an HTTP response body
+  #
+  # @param res_body [String] HTTP response body obtained via a GET request to the ADAudit Plus base path
+  # @return [Array] Array containing a status code and status message. If AD domain aliases were found, the Array also contains an Array of Strings for the domain aliases
+  def adaudit_plus_grab_domain_aliases(res_body)
+    domain_info = res_body.scan(/<select id="domainName" name="domainName" tabindex="\d+".*?>(.*?)<\/select>/m)&.flatten&.first
+    domain_aliases = domain_info&.scan(/<option value="(.*?)">.*?<\/option>/m)&.flatten
+    if domain_aliases.blank? || !domain_aliases.kind_of?(Array) || domain_aliases == ['ADAuditPlus Authentication']
+      return [adaudit_plus_status::NO_MATCH, 'No configured active directory domains were found.']
+    end
+
+    # remove 'ADAuditPlus Authentication' from the configured domains if it is listed
+    domain_aliases.delete('ADAuditPlus Authentication')
+
+    [adaudit_plus_status::SUCCESS, "Identified #{domain_aliases.length} configured domain alias(es): #{domain_aliases.join(', ')}", domain_aliases]
+  end
+
+  # Performs an API call to obtain the configured domains. The adapcsrf cookie obtained from this request is necessary to perform further authenticated actions
+  #
+  # @param adapcsrf_cookie [String] A valid adapcsrf_cookie obtained via a successful login action
+  # @param silent [Boolean] Whether to run in silent mode or not (silent mode means fewer print messages)
+  # @return [Hash] Hash containing keys of type String that are mapped to a status code (Integer), a status message (String), a cookie (String) and/or an Array of configured domains as Strings.
```

Comment:
You know what the hash key names are so this description should be updated to mention them in my opinion.